components/tidb_query/src/codec/batch/lazy_column.rs

```diff
@@ -215,6 +215,21 @@ impl LazyBatchColumn {
             LazyBatchColumn::Decoded(ref v) => v.encode(row_index, field_type, output),
         }
     }
+
+    pub fn append_to_chunk(
```

Comment:
I would more prefer adding `append_vec` to the `Chunk`, since the codec layer should be above the basic data structures and this component should not *depend on* the `Chunk`. Note that current code base does not do it right. But I don't think it is a good idea for new code base to do it incorrectly as well.

Reply:
Update. PTAL again.